aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/TestConfiguration.java

```diff
@@ -0,0 +1,125 @@
+ */
+
+package com.amplifyframework.storage.s3;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.test.core.app.ApplicationProvider;
+
+import com.amplifyframework.AmplifyException;
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.AmplifyConfiguration;
+import com.amplifyframework.core.InitializationStatus;
+import com.amplifyframework.core.category.CategoryType;
+import com.amplifyframework.hub.HubChannel;
+import com.amplifyframework.testutils.Await;
+
+import com.amazonaws.mobile.client.AWSMobileClient;
+import com.amazonaws.mobile.client.Callback;
+import com.amazonaws.mobile.client.UserStateDetails;
+import org.json.JSONException;
+
+final class TestConfiguration {
+    private static TestConfiguration singleton;
+    private final AWSS3StoragePlugin plugin;
+    private final String bucket;
+
+    private TestConfiguration(Context context) throws AmplifyException {
+        plugin = new AWSS3StoragePlugin();
+        bucket = getBucketNameFromPlugin(context, plugin);
+
+        Amplify.addPlugin(plugin);
+        configureAmplify(context);
+        setUpCredentials(context);
+    }
+
+    /**
+     * Process-wide configuration for the Storage instrumentation tests.
+     * @return A TestConfiguration instance
+     */
+    @NonNull
+    static synchronized TestConfiguration configureIfNotConfigured() throws AmplifyException {
+        if (singleton == null) {
+            singleton = new TestConfiguration(ApplicationProvider.getApplicationContext());
+        }
+        return singleton;
+    }
+
+    private static void configureAmplify(Context context) {
+        // Configure Amplify, and wait for Storage to be ready.
+        Await.result((onResult, onError) -> {
+            // Listen for initialization success messages
+            Amplify.Hub.subscribe(HubChannel.STORAGE, event -> {
+                // When we get one, end the Await call by firing a result
+                if (InitializationStatus.SUCCEEDED.toString().equals(event.getName())) {
+                    onResult.accept(CategoryType.STORAGE);
+                    // When we see a failure, end the await by firing onError
+                } else if (InitializationStatus.FAILED.toString().equals(event.getName())) {
+                    onError.accept(new RuntimeException(String.valueOf(event.getData())));
+                }
+            });
+            // Now that we're listening for it ... configure Amplify and begin initialization
+            try {
+                Amplify.configure(context);
+            } catch (AmplifyException configurationFailure) {
+                // If the configuration fails before even initialization begins, kill the Await with onError.
+                onError.accept(new RuntimeException("Configuration failed.",
+                        configurationFailure));
+            }
+        });
+    }
+
+    private static void setUpCredentials(Context context) {
```

Comment:
We have this code repeated enough times in our code base now that we shuold make a `TestMobileClient` utility or something like that (if we don't already have one??) So that we can just do like:
```
SynchronousMobileClient.initialize();
```
And have that work in a blocking way.